algorithms/Python/searching/interpolation_search.py

```diff
@@ -0,0 +1,39 @@
+def interpolation_search(array, x):
+    """
+    Use interpolation search to find a specified integer
+    """
+    if isinstance(x, int) and isinstance(array, list): # check that x is integer and array is list
+        if len(array) < 1: # if array length is less than 1 print message
+            return False
+        elif len(array) == 1 and x in array: # if x is only element of array return index 0
+            return 0
+        elif len(array) > 1 and x in array: # search if array length > 1 and x is in the array
+            min = 0
+            max = len(array) - 1
+            while min <= max:
+                p = int(min + (x-array[min])*(max-min) / (array[max]-array[min]))
```

Comment:
This formula will fail in the case when all the elements of the array are same ( i.e. `array[max]` == `array[min]` ) with  a `ZeroDivisionError`.

Reply:
I have updated this

Reply:
Shouldn't the output be consistant ?

the output of your new special case is (`min, max`) which is inconsistent compared to any other test case where it just returns only one value at where the search value is to be found.

Reply:
but if array[max]==array[min] then there are two elements with the target value, so i thought i should return both indices. Is it better to only return one index?

Reply:
consider the following cases 

---
**Case 1 :** 
```
arr = [1,1,1,1,1,1]
target_value = 1
```
your function returns `(0,5)`

---
**Case 2 :** 
```
arr = [0,0,0,1,1,1,1,1,1,2,3,4,5]
target_value = 1
```
your function returns `3`

---

You can see that although the answer is correct in both the cases, it shows inconsistency in output where for one exceptional case, the function is returning more than one values and for other it isn't.

either make the function return only one value, or handle the duplicates and return all the instances of the value found in the array ( both are fairly trivial tasks )

Reply:
Ok I have made it only return the min index, is this done now?